include/services/daal_atomic_int.h

```diff
@@ -37,68 +40,293 @@ namespace interface1
  * @ingroup memory
  * @{
  */
+
+#if defined(_WIN32) && !defined(_WIN64)
+/**
+ * <a name="DAAL-CLASS-SERVICES__ATOMIC"></a>
+ * \brief Class that represents an atomic object
+ *
+ * \tparam dataType Data type of the atomic object
+ */
+template <typename dataType>
+class DAAL_EXPORT Atomic
+{
+public:
+    /**
+     * Returns an increment of atomic object
+     * \return An increment of atomic object
+     */
+    inline dataType inc()
+    {
+        DAAL_ASSERT(sizeof(my_storage) == sizeof(long))
```

Comment:
If the class to be moved into cpp, it can be replaced by static_assert (due to c++11 usage).

Reply:
I don`t plan to move it into cpp